wp-content/themes/borderzine/inc/widgets/class-borderzine-3-col-widget.php

```diff
@@ -0,0 +1,388 @@
+	 * @global $wp_query Used to get posts on the page not in $shown_ids, to avoid duplicating posts
+	 */
+	public function widget( $args, $instance ) {
+
+		global $post,
+			$wp_query, // grab this to copy posts in the main column
+			$shown_ids; // an array of post IDs already on a page so we can avoid duplicating posts;
+
+		// Preserve global $post
+		$preserve = $post;
+
+
+		// Add the link to the title.
+		$title = apply_filters( 'widget_title', empty( $instance['title'] ) ? '' : $instance['title'], $instance, $this->id_base );
+
+		echo wp_kses_post( $args['before_widget'] );
+
+		if ( ! empty( $title ) ) {
+			echo $args['before_title'] . wp_kses_post( $title ). $args['after_title'];
+		}
+
+		$thumb = isset( $instance['thumbnail_display'] ) ? $instance['thumbnail_display'] : 'small';
+		$excerpt = isset( $instance['excerpt_display'] ) ? $instance['excerpt_display'] : 'num_sentences';
+
+		$query_args = array (
+			'post__not_in'   => get_option( 'sticky_posts' ),
+			'posts_per_page' => isset( $instance['num_posts'] ) ? $instance['num_posts'] : 3,
+			'post_status'    => 'publish',
+			'tax_query'      => array(),
+		);
+
+		if ( isset( $instance['avoid_duplicates'] ) && 1 === $instance['avoid_duplicates'] ) {
+			$query_args['post__not_in'] = $shown_ids;
+		}
+		if ( ! empty( $instance['cat'] ) ) {
+			$query_args['cat'] = $instance['cat'];
+		}
+		if ( ! empty( $instance['tag'] ) ) {
+			$query_args['tag'] = $instance['tag'];
+		}
+		if ( ! empty( $instance['author'] ) ) {
+			$query_args['author'] = $instance['author'];
+		}
+		if ( ! empty( $instance['prominence'] ) ) {
+			$query_args['tax_query'] = array_merge(
+				$query_args['tax_query'],
+				array(
+					array(
+						'taxonomy' => 'prominence',
+						'field'    => 'term_id',
+						'terms'    => $instance['prominence'],
+					),
+				)
+			);
+		}
+		if ( ! empty( $instance['taxonomy'] ) && ! empty( $instance['term'] ) ) {
+			$query_args['tax_query'] = array_merge(
+				$query_args['tax_query'],
+				array(
+					array(
+						'taxonomy' => $instance['taxonomy'],
+						'field'    => 'slug',
+						'terms'    => $instance['term'],
+					),
+				)
+			);
+		}
+
+		echo '<ul>';
+
+		$my_query = new WP_Query( $query_args );
+
+		if ( $my_query->have_posts() ) {
+
+			$output = '';
+
+			while ( $my_query->have_posts() ) {
+				$my_query->the_post();
+				$shown_ids[] = get_the_ID();
+
+				// wrap the items in li's.
+				$output .= sprintf(
+					'<li class="%1$s" >',
+					implode( ' ', get_post_class( '', get_the_id() ) )
+				);
+
+				$context = array(
+					'instance' => $instance,
+					'thumb' => $thumb,
+					'excerpt' => $excerpt,
+				);
+
+				ob_start();
+				largo_render_template( 'partials/widget', 'content', $context );
+				$output .= ob_get_clean();
+
+				// close the item
+				$output .= '</li>';
+
+			} // endwhile.
+
+			// print all of the items
+			echo $output;
+
+		} else {
+			printf(
+				'<p class="error"><strong>%1$s</strong></p>',
+				sprint(
```

Comment:
@benlk This should be `sprintf`.

Reply:
Fixed in https://github.com/INN/umbrella-borderzine/pull/67/commits/fadca48445f4f7bc95a941cdfd05f57fd3b16be5